Modules/mmapmodule.c

```diff
@@ -692,11 +692,51 @@ mmap__sizeof__method(mmap_object *self, void *unused)
 }
 #endif
 
+#ifdef HAVE_MADVISE
+static PyObject *
+mmap_madvise_method(mmap_object *self, PyObject *args)
+{
+    int option;
+    Py_ssize_t start = 0, length;
+
+    CHECK_VALID(NULL);
+    length = self->size;
+
+    if (!PyArg_ParseTuple(args, "i|nn:madvise", &option, &start, &length)) {
+        return NULL;
+    }
+
+    if (start < 0 || start > self->size) {
+        PyErr_SetString(PyExc_ValueError, "madvise start invalid");
+        return NULL;
+    }
+
+    if (length <= 0 || length > self->size) {
+        PyErr_SetString(PyExc_ValueError, "madvise length invalid");
+        return NULL;
+    }
+
+    if (start + length > self->size) {
```

Comment:
I understand this is necessary if `start` is passed but not `length`, but it seems a bit lenient compared to the checks above where we refuse `length > self->size`. How about tweaking the length default value instead (you could perhaps initialize it to `Py_SSIZE_T_MAX`).